fastai/vision/learner.py

```diff
@@ -95,119 +96,114 @@ def unet_learner(data:DataBunch, arch:Callable, pretrained:bool=True, blur_final
-        plt.figure(**kwargs)
-        plt.imshow(cm, interpolation='nearest', cmap=cmap)
-        plt.title(title)
-        tick_marks = arange_of(self.data.classes)
-        plt.xticks(tick_marks, self.data.classes, rotation=90)
-        plt.yticks(tick_marks, self.data.classes, rotation=0)
-
-        if normalize: cm = cm.astype('float') / cm.sum(axis=1)[:, np.newaxis]
-        thresh = cm.max() / 2.
-        for i, j in itertools.product(range(cm.shape[0]), range(cm.shape[1])):
-            coeff = f'{cm[i, j]:.{norm_dec}f}' if normalize else f'{cm[i, j]}'
-            plt.text(j, i, coeff, horizontalalignment="center", color="white" if cm[i, j] > thresh else "black")
-
-        plt.tight_layout()
-        plt.ylabel('Actual')
-        plt.xlabel('Predicted')
-
-    def most_confused(self, min_val:int=1, slice_size:int=None)->Collection[Tuple[str,str,int]]:
-        "Sorted descending list of largest non-diagonal entries of confusion matrix, presented as actual, predicted, number of occurrences."
-        cm = self.confusion_matrix(slice_size=slice_size)
-        np.fill_diagonal(cm, 0)
-        res = [(self.data.classes[i],self.data.classes[j],cm[i,j])
-                for i,j in zip(*np.where(cm>min_val))]
-        return sorted(res, key=itemgetter(2), reverse=True)
-
-def _learner_interpret(learn:Learner, ds_type:DatasetType=DatasetType.Valid, tta=False):
-    "Create a `ClassificationInterpretation` object from `learner` on `ds_type` with `tta`."
-    return ClassificationInterpretation.from_learner(learn, ds_type=ds_type, tta=tta)
+    "Plot the confusion matrix, with `title` and using `cmap`."
+    # This function is mainly copied from the sklearn docs
+    cm = self.confusion_matrix(slice_size=slice_size)
+    plt.figure(**kwargs)
+    plt.imshow(cm, interpolation='nearest', cmap=cmap)
+    plt.title(title)
+    tick_marks = arange_of(self.data.classes)
+    plt.xticks(tick_marks, self.data.classes, rotation=90)
+    plt.yticks(tick_marks, self.data.classes, rotation=0)
+
+    if normalize: cm = cm.astype('float') / cm.sum(axis=1)[:, np.newaxis]
+    thresh = cm.max() / 2.
+    for i, j in itertools.product(range(cm.shape[0]), range(cm.shape[1])):
+        coeff = f'{cm[i, j]:.{norm_dec}f}' if normalize else f'{cm[i, j]}'
+        plt.text(j, i, coeff, horizontalalignment="center", color="white" if cm[i, j] > thresh else "black")
+
+    plt.tight_layout()
+    plt.ylabel('Actual')
+    plt.xlabel('Predicted')
+
+def most_confused(self, min_val:int=1, slice_size:int=None)->Collection[Tuple[str,str,int]]:
+    "Sorted descending list of largest non-diagonal entries of confusion matrix."
+    cm = self.confusion_matrix(slice_size=slice_size)
+    np.fill_diagonal(cm, 0)
+    res = [(self.data.classes[i],self.data.classes[j],cm[i,j])
+            for i,j in zip(*np.where(cm>min_val))]
+    return sorted(res, key=itemgetter(2), reverse=True)
+
+ClassificationInterpretation.top_losses = top_losses
```

Comment:
Only `plot_top_losses` and `plot_multi_top_losses` need to be monkey-patched, they are the only true new methods. And `from_learner` if we want to add TTA.